### ISSUE TITLE
Cluster-autoscaler and Kube-scheduler differ with each other

### ISSUE THREAD

**Issue by rdelgaizo** [kubernetes/autoscaler#3867]
**Which component are you using?**: cluster-autoscaler

**What version of the component are you using?**: 

Component version: v1.18.1 (installed with helm chart 9.4.0)

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
</pre></details>

**What environment is this in?**:
AWS EKS 1.18

**What did you expect to happen?**:

I expected that cluster-autoscaler would increase the number of nodes in the auto-scale group within AWS when there was not enough resources available.

**What happened instead?**:

Kube-scheduler saw there there are no nodes with enough CPU to allocate the pod to. Cluster-autoscaler thought that there was enough space on a node to schedule it and would not scale up.

**How to reproduce it (as minimally and precisely as possible)**:

I used a bunch of pods all with requests of 100m each and the total on the node was 2000m. When then node reached 1900m allocated I ran into this issue.

**Anything else we need to know?**:

I believe this has to do with kube-scheduler using the Allocatable value for CPU, but cluster-autoscaler using the Capacity CPU. Because there was exactly 100m left over in the total capacity (but not allocatable), cluster-autoscaler thought it should fit, but kube-scheduler would not place it.

When I set my pods to request 101m instead, cluster-autoscaler then scaled up as expected.

**Node describe:**
```
Capacity:
  attachable-volumes-aws-ebs:  25
  cpu:                         2
  ephemeral-storage:           62902252Ki
  hugepages-1Gi:               0
  hugepages-2Mi:               0
  memory:                      7937472Ki
  pods:                        29
Allocatable:
  attachable-volumes-aws-ebs:  25
  cpu:                         1930m
  ephemeral-storage:           56896973524
  hugepages-1Gi:               0
  hugepages-2Mi:               0
  memory:                      7247296Ki
  pods:                        29
```

**Allocated resources:**
```
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource                    Requests          Limits
  --------                    --------          ------
  cpu                         1900m (98%)       8 (414%)
  memory                      4549824896 (61%)  26069803776 (351%)
```

**Log output from cluster-autoscaler:**
```
I0202 14:42:33.466769       1 filter_out_schedulable.go:106] Pod default.indexer-847856f5c-w8xgh marked as unschedulable can be scheduled on node ip-10-10-10.10.ec2.internal. Ignoring in scale up.",
```

**Log from kube-scheduler:**
```
I0202 14:36:51.840257       1 factory.go:462] Unable to schedule default/ndexer-847856f5c-w8xgh: no fit: 0/5 nodes are available: 2 node(s) didn't match node selector, 3 Insufficient cpu.; waiting
--
E0202 14:36:51.845599       1 factory.go:503] pod: default/indexer-847856f5c-w8xgh is already present in unschedulable queue
```

**Chart settings for extraArgs:**
```
        extraArgs:
          balance-similar-node-groups: true
          expander: "least-waste"
          logtostderr: true
          stderrthreshold: "info"
          v: 4
```
I did also try this removing expander and setting balance-similar-node-groups to false and saw the same issue.

**Deployment resource block:**
```
          resources:
            limits:
              cpu: "500m"
              memory: "1536Mi"
            requests:
              cpu: "100m"
              memory: "256Mi"
```

**Comment by rdelgaizo** [comment]
I have now also seen this come up when autoscaler should not think it can be scheduled by a lot. The pod in question is requesting 101m CPU.

```
Pod default.indexer-6b49cf6797-6wwp8 marked as unschedulable can be scheduled on node ip-10-10-10-99.ec2.internal. Ignoring in scale up.
```

**kubectl describe node ip-10-10-10-99.ec2.internal**:
```
Capacity:
  cpu:                         2
Allocatable:
  cpu:                         1930m
System Info:
  Kernel Version:             5.4.80-40.140.amzn2.x86_64
  OS Image:                   Amazon Linux 2
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  docker://19.3.6
  Kubelet Version:            v1.18.9-eks-d1db3c
  Kube-Proxy Version:         v1.18.9-eks-d1db3c            
Non-terminated Pods:          (20 in total)
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource                    Requests          Limits
  --------                    --------          ------
  cpu                         1910m (98%)       6 (310%)
  memory                      3774518528 (50%)  17559869184 (236%)
  ephemeral-storage           0 (0%)            0 (0%)
  hugepages-1Gi               0 (0%)            0 (0%)
  hugepages-2Mi               0 (0%)            0 (0%)
  attachable-volumes-aws-ebs  0                 0
Events:                       <none>
```

**Comment by harshal-shah** [comment]
I can confirm a similar problem in our setup as well. 
We have multiple node groups and we use priority expander with a higher precedence given to spot-instance node groups. 
In our case, some spot instances were interrupted and we saw the message 

~~~
I0304 20:14:36.581309       1 filter_out_schedulable.go:106] Pod masked-ns.masked-service-dk-7549b7c68d-ww5p6 marked as unschedulable can be scheduled on node template-node-for-platform-staging-eks-node-lt-4x16-spot20200428084500418300000004-2155088869513297691-3. Ignoring in scale up.
~~~

Autoscaler should have scaled up other ASG's but it just ignored scale up till the 4x16 spot nodes got added back to our cluster

![image](https://user-images.githubusercontent.com/16479600/110078898-af007880-7d88-11eb-9491-0921d7945e21.png)
